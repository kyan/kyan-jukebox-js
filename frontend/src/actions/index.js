import MopidyApi from 'constants/mopidy-api'
import Types from 'constants/common'
import { transformUrl } from 'utils/spotify'

export const validateUser = (email, user) => {
  return {
    type: Types.VALIDATE_USER,
    email,
    user
  }
}

export const updateUser = (email, user) => {
  return {
    type: Types.STORE_USER,
    email,
    user
  }
}

export const clearUser = () => {
  return {
    type: Types.CLEAR_USER
  }
}

export const setAuthError = error => {
  return {
    type: Types.SET_AUTH_ERROR,
    error
  }
}

export const validateUserRequest = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.VALIDATE_USER
  }
}

export const addNewTrack = url => {
  let uri = transformUrl(url)

  return {
    type: Types.SEND,
    key: MopidyApi.TRACKLIST_ADD_TRACK,
    params: { uris: [uri] }
  }
}

export const addNewTracks = uris => {
  return {
    type: Types.SEND,
    key: MopidyApi.TRACKLIST_ADD_TRACK,
    params: { uris: uris.map(uri => transformUrl(uri)) }
  }
}

export const addCurrentTrack = track => {
  return {
    type: Types.ADD_CURRENT_TRACK,
    track
  }
}

export const syncSocialData = track => {
  return {
    type: Types.SYNC_SOCIAL_DATA,
    track
  }
}

export const addTrackList = tracks => {
  return {
    type: Types.ADD_TRACKS,
    tracks
  }
}

export const removeFromTracklist = uri => {
  return {
    type: Types.SEND,
    key: MopidyApi.TRACKLIST_REMOVE_TRACK,
    params: { criteria: { uri: [uri] } }
  }
}

export const updateProgressTimer = (position, duration) => {
  if (duration === Infinity) {
    duration = 0
  }

  return {
    type: Types.UPDATE_PROGRESS_TIMER,
    position,
    duration
  }
}

export const wsConnect = () => {
  return {
    type: Types.CONNECT
  }
}

export const wsConnecting = () => {
  return {
    type: Types.CONNECTING
  }
}

export const wsConnected = () => {
  return {
    type: Types.CONNECTED
  }
}

export const wsDisconnect = () => {
  return {
    type: Types.DISCONNECT
  }
}

export const wsDisconnected = () => {
  return {
    type: Types.DISCONNECTED
  }
}

export const mopidyConnected = () => {
  return {
    type: Types.MOPIDY_CONNECTED
  }
}

export const mopidyDisconnected = () => {
  return {
    type: Types.MOPIDY_DISCONNECTED
  }
}

export const getCurrentTrack = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_GET_CURRENT_TRACK
  }
}

export const getTimePosition = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_GET_TIME_POSITION
  }
}

export const getState = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_GET_PLAYBACK_STATE
  }
}

export const getTrackList = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.TRACKLIST_GET_TRACKS
  }
}

export const clearTrackList = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.TRACKLIST_CLEAR
  }
}

export const startPlaying = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_PLAY
  }
}

export const stopPlaying = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_STOP
  }
}

export const pausePlaying = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_PAUSE
  }
}

export const nextPlaying = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_NEXT
  }
}

export const previousPlaying = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.PLAYBACK_BACK
  }
}

export const getVolume = () => {
  return {
    type: Types.SEND,
    key: MopidyApi.GET_VOLUME
  }
}

export const updateVolume = volume => {
  return {
    type: Types.UPDATE_VOLUME,
    volume
  }
}

export const updatePlaybackState = state => {
  return {
    type: Types.UPDATE_PLAYBACK_STATE,
    state
  }
}

export const setVolume = volume => {
  return {
    type: Types.SEND,
    key: MopidyApi.SET_VOLUME,
    params: [Number(volume)]
  }
}
