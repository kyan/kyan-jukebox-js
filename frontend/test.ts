#!/usr/bin/env bun
/**
 * Simplified Bun Test Runner with Jest Compatibility
 *
 * Uses Bun's Jest compatibility mode to run existing Jest-style tests
 * without requiring complex migration of test files.
 */

import { spawnSync } from 'child_process'
import { existsSync } from 'fs'

interface TestOptions {
  watch?: boolean
  coverage?: boolean
  verbose?: boolean
  silent?: boolean
  bail?: boolean
  watchAll?: boolean
}

class BunJestRunner {
  private options: TestOptions = {}

  constructor(args: string[] = []) {
    this.parseArgs(args)
  }

  private parseArgs(args: string[]): void {
    for (let i = 0; i < args.length; i++) {
      const arg = args[i]

      switch (arg) {
        case '--watch':
        case '-w':
          this.options.watch = true
          break
        case '--watchAll':
          this.options.watchAll = true
          break
        case '--watchAll=false':
          this.options.watchAll = false
          this.options.watch = false
          break
        case '--coverage':
          this.options.coverage = true
          break
        case '--verbose':
          this.options.verbose = true
          break
        case '--silent':
          this.options.silent = true
          break
        case '--bail':
          this.options.bail = true
          break
        case '--forceExit':
        case '--detectOpenHandles':
        case '--maxWorkers=1':
          // Legacy Jest options - ignore as Bun handles these automatically
          break
      }
    }

    // Default watch behavior
    if (
      !process.env.CI &&
      this.options.watchAll === undefined &&
      this.options.watch === undefined
    ) {
      this.options.watch = true
    }
  }

  private setupEnvironment(): void {
    // Set test environment variables
    process.env.NODE_ENV = 'test'
    process.env.BABEL_ENV = 'test'
    process.env.PUBLIC_URL = ''
  }

  private async runPreflightChecks(): Promise<void> {
    if (this.options.silent) return

    console.log('üß™ Running preflight checks...\n')

    // Type checking
    if (existsSync('./tsconfig.json')) {
      console.log('üìù Type checking...')
      try {
        const typeCheck = spawnSync('bun', ['run', 'type-check'], {
          stdio: 'inherit'
        })

        if (typeCheck.status !== 0) {
          console.error('‚ùå Type check failed')
          process.exit(1)
        } else {
          console.log('‚úÖ Type check passed')
        }
      } catch (error) {
        console.warn('‚ö†Ô∏è Type checking skipped:', error.message)
      }
    }

    // Linting
    console.log('üîç Linting...')
    try {
      const lint = spawnSync('bun', ['run', 'lint'], {
        stdio: 'inherit'
      })

      if (lint.status !== 0) {
        console.error('‚ùå Linting failed')
        process.exit(1)
      } else {
        console.log('‚úÖ Linting passed')
      }
    } catch (error) {
      console.warn('‚ö†Ô∏è Linting skipped:', error.message)
    }

    // Format check
    console.log('üíÖ Format checking...')
    try {
      const format = spawnSync('bun', ['run', 'format:check'], {
        stdio: 'inherit'
      })

      if (format.status !== 0) {
        console.error('‚ùå Format check failed')
        console.log('üí° Run: bun run format')
        process.exit(1)
      } else {
        console.log('‚úÖ Format check passed')
      }
    } catch (error) {
      console.warn('‚ö†Ô∏è Format checking skipped:', error.message)
    }

    console.log('')
  }

  private buildJestCommand(): { command: string; args: string[] } {
    const args: string[] = []

    // Test pattern - Jest will find spec files automatically
    args.push('--testMatch', '<rootDir>/src/**/*.{spec,test}.{js,jsx,ts,tsx}')

    // Watch mode
    if (this.options.watchAll) {
      args.push('--watchAll')
    } else if (this.options.watch) {
      args.push('--watch')
    } else {
      args.push('--watchAll=false')
    }

    // Coverage
    if (this.options.coverage) {
      args.push('--coverage')
    }

    // Verbose mode
    if (this.options.verbose) {
      args.push('--verbose')
    }

    // Silent mode
    if (this.options.silent) {
      args.push('--silent')
    }

    // Bail on first failure
    if (this.options.bail) {
      args.push('--bail')
    }

    // Force exit and handle detection for CI
    if (process.env.CI || !this.options.watch) {
      args.push('--forceExit', '--detectOpenHandles', '--maxWorkers=1')
    }

    return { command: 'bunx', args: ['jest', ...args] }
  }

  async run(): Promise<void> {
    if (!this.options.silent) {
      console.log('üèÉ‚Äç‚ôÇÔ∏è Bun Jest Test Runner\n')
    }

    // Setup test environment
    this.setupEnvironment()

    // Run preflight checks (validation)
    await this.runPreflightChecks()

    // Build and run Jest command through Bun
    const { command, args } = this.buildJestCommand()

    if (this.options.verbose && !this.options.silent) {
      console.log('üìã Command:', [command, ...args].join(' '))
    }

    if (!this.options.silent) {
      console.log('üß™ Running tests...\n')
    }

    try {
      const result = spawnSync(command, args, {
        stdio: 'inherit',
        cwd: process.cwd(),
        env: {
          ...process.env,
          // Ensure Bun uses Jest compatibility mode
          FORCE_COLOR: '1'
        }
      })

      if (result.status !== 0) {
        if (!this.options.silent) {
          console.error('\n‚ùå Tests failed')
        }
        process.exit(result.status || 1)
      } else {
        if (!this.options.silent) {
          console.log('\n‚úÖ All tests passed!')
        }
      }
    } catch (error) {
      console.error('‚ùå Test runner failed:', error.message)

      if (!this.options.silent) {
        console.log('\nüí° Troubleshooting:')
        console.log('   ‚Ä¢ Ensure Bun v1.0+ is installed: bun --version')
        console.log('   ‚Ä¢ Try: bun install (to ensure dependencies are correct)')
        console.log('   ‚Ä¢ Check Jest configuration in package.json')
      }

      process.exit(1)
    }
  }
}

// CLI execution
if (import.meta.main) {
  const args = process.argv.slice(2)

  if (args.includes('--help') || args.includes('-h')) {
    console.log(`
Bun Jest Test Runner

Usage:
  bun test.ts [options]

Options:
  --watch, -w          Watch files for changes
  --watchAll           Watch all files
  --watchAll=false     Disable watch mode (for CI)
  --coverage           Generate coverage report
  --verbose            Verbose output
  --silent             Suppress output except errors
  --bail               Stop on first test failure

Examples:
  bun test.ts                           # Run all tests (watch mode in dev)
  bun test.ts --watchAll=false         # CI mode - run once
  bun test.ts --coverage               # With coverage report
  bun test.ts --silent --watchAll=false # Silent CI mode
`)
    process.exit(0)
  }

  const runner = new BunJestRunner(args)
  runner.run().catch(error => {
    console.error('Test runner failed:', error)
    process.exit(1)
  })
}
