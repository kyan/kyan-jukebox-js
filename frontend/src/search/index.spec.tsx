import { describe, it, expect, beforeEach, mock } from 'bun:test'
import React from 'react'
import { Provider } from 'react-redux'
import { render, fireEvent } from '@testing-library/react'
import configureMockStore from 'redux-mock-store'
import { toggleSearchSidebar } from './actions'
import SearchContainer from './index'

describe('SearchContainer', () => {
  const mockStore = configureMockStore()

  beforeEach(() => {
    // Bun mocks are cleared automatically between tests
  })

  describe('render', () => {
    it('Adding a track works as expected', () => {
      const store = mockStore({
        search: {
          activePage: 2,
          limit: 20,
          query: 'happy',
          total: 3200,
          results: [
            {
              name: 'Track name 1',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack1',
              artist: {
                name: 'Artist name 1'
              },
              album: {
                name: 'Album name 1'
              },
              image: 'image1'
            }
          ],
          searchSideBarOpen: true
        },
        curatedList: {
          tracks: []
        }
      })
      const { getByText, getByAltText } = render(
        <Provider store={store}>
          <SearchContainer />
        </Provider>
      )

      fireEvent.click(getByText('Find'))
      fireEvent.click(getByAltText('Track name 1'))
      fireEvent.click(getByText('2'))

      const actions = store.getActions()

      // Should dispatch search action when Find button is clicked
      expect(actions).toEqual([
        {
          key: 'searchGetTracks',
          params: {
            options: {
              limit: 20,
              offset: 0
            },
            query: 'happy'
          },
          type: 'actionTrackSearch'
        },
        {
          type: 'actionRemoveFromSearchResults',
          uris: ['https://open.spotify.com/track/0c41pMosF5Kqwwetrack1']
        },
        {
          key: 'tracklist.add',
          params: {
            uris: ['spotify:track:0c41pMosF5Kqwwetrack1']
          },
          type: 'actionSend'
        },
        {
          key: 'searchGetTracks',
          params: {
            options: {
              limit: 20,
              offset: 20
            },
            query: 'happy'
          },
          type: 'actionTrackSearch'
        }
      ])
    })

    it('Adding a track to the mix works as expected', () => {
      const store = mockStore({
        search: {
          activePage: 2,
          limit: 20,
          query: 'happy',
          total: 3200,
          results: [
            {
              name: 'Track name 1',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack1',
              artist: {
                name: 'Artist name 1'
              },
              album: {
                name: 'Album name 1'
              },
              image: 'image1'
            }
          ],
          searchSideBarOpen: true
        },
        curatedList: {
          tracks: []
        }
      })
      const { getAllByText } = render(
        <Provider store={store}>
          <SearchContainer />
        </Provider>
      )
      fireEvent.click(getAllByText('Add to mix')[0])

      const actions = store.getActions()
      expect(actions).toEqual([
        {
          type: 'actionRemoveFromSearchResults',
          uris: ['https://open.spotify.com/track/0c41pMosF5Kqwwetrack1']
        },
        {
          track: {
            album: {
              name: 'Album name 1'
            },
            artist: {
              name: 'Artist name 1'
            },
            image: 'image1',
            name: 'Track name 1',
            uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack1'
          },
          type: 'curateAddTrackToMix'
        }
      ])
    })

    it('Removing a track from the mix works as expected', () => {
      const store = mockStore({
        search: {
          activePage: 2,
          limit: 20,
          query: 'happy',
          total: 3200,
          results: [
            {
              name: 'Track name 1',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack1',
              artist: {
                name: 'Artist name 1'
              },
              album: {
                name: 'Album name 1'
              },
              image: 'image1'
            }
          ],
          searchSideBarOpen: true
        },
        curatedList: {
          tracks: [
            {
              name: 'Track name 2',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack2',
              artist: {
                name: 'Artist name 2'
              },
              album: {
                name: 'Album name 2'
              },
              image: 'image2'
            }
          ]
        }
      })
      const { getAllByText } = render(
        <Provider store={store}>
          <SearchContainer />
        </Provider>
      )
      fireEvent.click(getAllByText('Remove')[0])

      const actions = store.getActions()
      expect(actions).toEqual([
        {
          type: 'curateRemoveTracksFromMix',
          uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack2'
        },
        {
          key: 'searchGetTracks',
          params: {
            options: {
              limit: 20,
              offset: 0
            },
            query: 'happy'
          },
          type: 'actionTrackSearch'
        }
      ])
    })

    it('Adding tracks from the mix works as expected', () => {
      const store = mockStore({
        search: {
          activePage: 2,
          limit: 20,
          query: 'happy',
          total: 3200,
          results: [
            {
              name: 'Track name 1',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack1',
              artist: {
                name: 'Artist name 1'
              },
              album: {
                name: 'Album name 1'
              },
              image: 'image1'
            }
          ],
          searchSideBarOpen: true
        },
        curatedList: {
          tracks: [
            {
              name: 'Track name 2',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack2',
              artist: {
                name: 'Artist name 2'
              },
              album: {
                name: 'Album name 2'
              },
              image: 'image2'
            },
            {
              name: 'Track name 3',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack3',
              artist: {
                name: 'Artist name 3'
              },
              album: {
                name: 'Album name 3'
              },
              image: 'image3'
            }
          ]
        }
      })
      const { getByText } = render(
        <Provider store={store}>
          <SearchContainer />
        </Provider>
      )
      fireEvent.click(getByText('Add mix to playlist'))

      expect(store.getActions()).toEqual([
        {
          key: 'tracklist.add',
          params: {
            uris: ['spotify:track:0c41pMosF5Kqwwetrack2', 'spotify:track:0c41pMosF5Kqwwetrack3']
          },
          type: 'actionSend'
        },
        {
          type: 'searchClearMix'
        }
      ])
    })

    it('Swapping tracks from the mix works as expected', () => {
      const store = mockStore({
        search: {
          activePage: 2,
          limit: 20,
          query: 'happy',
          total: 3200,
          results: [],
          searchSideBarOpen: true
        },
        curatedList: {
          tracks: [
            {
              name: 'Track name 1',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack1',
              artist: {
                name: 'Artist name 1'
              },
              album: {
                name: 'Album name 1'
              },
              image: 'image1'
            },
            {
              name: 'Track name 2',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack1',
              artist: {
                name: 'Artist name 2'
              },
              album: {
                name: 'Album name 2'
              },
              image: 'image2'
            },
            {
              name: 'Track name 3',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack3',
              artist: {
                name: 'Artist name 3'
              },
              album: {
                name: 'Album name 3'
              },
              image: 'image3'
            }
          ]
        }
      })
      const { getAllByTitle } = render(
        <Provider store={store}>
          <SearchContainer />
        </Provider>
      )
      const _actions = store.getActions()
      const draggableOpNodeE = getAllByTitle('You can drag this to sort.')[2]
      const createBubbledEvent = (type: any, props = {}) => {
        const event = new Event(type, { bubbles: true })
        Object.assign(event, props)
        return event
      }

      draggableOpNodeE.dispatchEvent(
        createBubbledEvent('drop', {
          dataTransfer: {
            getData: mock(() => '1')
          }
        })
      )

      expect(store.getActions()).toEqual([
        {
          a: 2,
          b: 1,
          type: 'searchSwapTracks'
        }
      ])
    })

    it('Does not add when more than 5 already added', () => {
      const store = mockStore({
        search: {
          activePage: 2,
          limit: 20,
          query: 'happy',
          total: 3200,
          results: [
            {
              name: 'Track name 1',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack1',
              artist: {
                name: 'Artist name 1'
              },
              album: {
                name: 'Album name 1'
              },
              image: 'image1'
            }
          ],
          searchSideBarOpen: true
        },
        curatedList: {
          tracks: [
            {
              name: 'Track name 2',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack2',
              artist: {
                name: 'Artist name 2'
              },
              album: {
                name: 'Album name 2'
              },
              image: 'image2'
            },
            {
              name: 'Track name 3',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack3',
              artist: {
                name: 'Artist name 3'
              },
              album: {
                name: 'Album name 3'
              },
              image: 'image3'
            },
            {
              name: 'Track name 4',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack4',
              artist: {
                name: 'Artist name 4'
              },
              album: {
                name: 'Album name 4'
              },
              image: 'image4'
            },
            {
              name: 'Track name 5',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack5',
              artist: {
                name: 'Artist name 5'
              },
              album: {
                name: 'Album name 5'
              },
              image: 'image5'
            },
            {
              name: 'Track name 6',
              uri: 'https://open.spotify.com/track/0c41pMosF5Kqwwetrack6',
              artist: {
                name: 'Artist name 6'
              },
              album: {
                name: 'Album name 6'
              },
              image: 'image6'
            }
          ]
        }
      })
      const { getByText, getAllByText } = render(
        <Provider store={store}>
          <SearchContainer>
            <span>Close</span>
          </SearchContainer>
        </Provider>
      )
      store.dispatch(toggleSearchSidebar(true))
      fireEvent.click(getAllByText('Add to mix')[0])
      fireEvent.click(getByText('Close'))

      const actions = store.getActions()
      expect(actions).toEqual([
        {
          open: true,
          type: 'actionToggleSearchSidebar'
        },
        {
          open: false,
          type: 'actionToggleSearchSidebar'
        }
      ])
    })
  })
})
