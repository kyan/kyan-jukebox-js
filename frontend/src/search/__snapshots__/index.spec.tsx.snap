// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SearchContainer render Adding a track to the mix works as expected 1`] = `
[
  {
    "type": "actionRemoveFromSearchResults",
    "uris": [
      "https://open.spotify.com/track/0c41pMosF5Kqwwetrack1",
    ],
  },
  {
    "track": {
      "album": {
        "name": "Album name 1",
      },
      "artist": {
        "name": "Artist name 1",
      },
      "image": "image1",
      "name": "Track name 1",
      "uri": "https://open.spotify.com/track/0c41pMosF5Kqwwetrack1",
    },
    "type": "curateAddTrackToMix",
  },
]
`;

exports[`SearchContainer render Adding a track works as expected 1`] = `
[
  {
    "open": true,
    "type": "actionToggleSearchSidebar",
  },
  {
    "key": "searchGetTracks",
    "params": {
      "options": {
        "limit": 20,
        "offset": 0,
      },
      "query": "happy",
    },
    "type": "actionTrackSearch",
  },
  {
    "type": "actionRemoveFromSearchResults",
    "uris": [
      "https://open.spotify.com/track/0c41pMosF5Kqwwetrack1",
    ],
  },
  {
    "key": "tracklist.add",
    "params": {
      "uris": [
        "spotify:track:0c41pMosF5Kqwwetrack1",
      ],
    },
    "type": "actionSend",
  },
  {
    "query": "spam",
    "type": "actionStoreSearchQuery",
  },
  {
    "key": "searchGetTracks",
    "params": {
      "options": {
        "limit": 20,
        "offset": 20,
      },
      "query": "happy",
    },
    "type": "actionTrackSearch",
  },
]
`;

exports[`SearchContainer render Adding tracks from the mix works as expected 1`] = `
[
  {
    "key": "tracklist.add",
    "params": {
      "uris": [
        "spotify:track:0c41pMosF5Kqwwetrack2",
        "spotify:track:0c41pMosF5Kqwwetrack3",
      ],
    },
    "type": "actionSend",
  },
  {
    "type": "searchClearMix",
  },
]
`;

exports[`SearchContainer render Does not add when more than 5 already added 1`] = `
[
  {
    "open": true,
    "type": "actionToggleSearchSidebar",
  },
  {
    "open": false,
    "type": "actionToggleSearchSidebar",
  },
]
`;

exports[`SearchContainer render Removing a track from the mix works as expected 1`] = `
[
  {
    "type": "curateRemoveTracksFromMix",
    "uri": "https://open.spotify.com/track/0c41pMosF5Kqwwetrack2",
  },
  {
    "key": "searchGetTracks",
    "params": {
      "options": {
        "limit": 20,
        "offset": 0,
      },
      "query": "happy",
    },
    "type": "actionTrackSearch",
  },
]
`;

exports[`SearchContainer render Swapping tracks from the mix works as expected 1`] = `
[
  {
    "a": 2,
    "b": 1,
    "type": "searchSwapTracks",
  },
]
`;
